File: lucafontanili.designpatterns.java/src/main/java/com/lucafontanili/designpatterns/behavioral/observer/ObserverClient.java
package com.lucafontanili.designpatterns.behavioral.observer;

import java.io.IOException;

import com.lucafontanili.designpatterns.utilities.AbstractClient;
import com.lucafontanili.designpatterns.utilities.InvalidArgumentException;

public class ObserverClient extends AbstractClient {

    public static void main(String[] args) throws InvalidArgumentException, IOException {
	main(new ObserverClient());
    }

    @Override
    public void run() throws InvalidArgumentException, IOException {
	ClockTimer clockTimer = new ClockTimer();
	clockTimer.addObserver(new ClockObserver12());
	clockTimer.addObserver(new ClockObserver24());
	for (int count = 0; count < 5; ++count) {
	    clockTimer.tick();
	    try {
		Thread.sleep(1000);
	    } catch (InterruptedException e) {
		LOGGER.error(new StringBuilder(32).append("Error during sleep"), e);
	    }
	}
    }

}

File: lucafontanili.designpatterns.python/src/behavioral/observer/ObserverClient.py
'''
Created on Nov 13, 2016

@author: Admin
'''

from ClockTimer import *
import time

clock_timer = ClockTimer()
clock_timer.add_observer(ClockObserver12())
clock_timer.add_observer(ClockObserver24())

for i in range(0,5):
    clock_timer.tick()
    time.sleep(1)